server/graphql/v2/object/Host.js

```diff
@@ -437,6 +472,236 @@ export const Host = new GraphQLObjectType({
+            collectiveIds = collectives.map(collective => collective.id);
+            where.CollectiveId = { [Op.in]: collectiveIds };
+          }
+          const expensesCount = await models.Transaction.count({
+            where,
+          });
+          let expenseAmountOverTime;
+          if (args.timeUnit) {
+            const dateFrom = args.dateFrom ? moment(args.dateFrom).toISOString() : undefined;
+            const dateTo = args.dateTo ? moment(args.dateTo).toISOString() : undefined;
+            expenseAmountOverTime = await sequelize.query(
+              `SELECT DATE_TRUNC(:timeUnit, "createdAt") AS "date", sum(amount) AS "amount", "currency"
+                FROM "Transactions" WHERE kind = :transactionKind AND "HostCollectiveId" = :hostCollectiveId
+                AND type = :transactionType
+                ${collectiveIds ? `AND "CollectiveId" IN (:collectiveIds)` : ``}
+                ${dateFrom ? `AND "createdAt" >= :dateFrom` : ``} ${dateTo ? `AND "createdAt" <= :dateTo` : ``}
+                GROUP BY DATE_TRUNC(:timeUnit, "createdAt"), "currency"
+                ORDER BY DATE_TRUNC(:timeUnit, "createdAt")`,
+              {
+                type: sequelize.QueryTypes.SELECT,
+                replacements: {
+                  hostCollectiveId: host.id,
+                  timeUnit: args.timeUnit,
+                  transactionKind: TransactionKind.EXPENSE,
+                  transactionType: TransactionTypes.DEBIT,
+                  collectiveIds,
+                  dateFrom,
+                  dateTo,
+                },
+              },
+            );
+            expenseAmountOverTime = expenseAmountOverTime.map(expenseAmount =>
+              convertCurrencyAmount(expenseAmount.currency, host.currency, expenseAmount.date, expenseAmount.amount),
+            );
+            expenseAmountOverTime = await Promise.all(expenseAmountOverTime);
+
+            const counts = expenseAmountOverTime.reduce((prev, curr) => {
+              const count = prev.get(curr.date) || 0;
+              prev.set(curr.date.toISOString(), curr.amount + count);
+              return prev;
+            }, new Map());
+
+            expenseAmountOverTime = [...counts].map(([date, amount]) => {
+              return { date, amount: Math.abs(amount), currency: host.currency };
+            });
+          }
+
+          expenseAmountOverTime = {
+            dateFrom: args.dateFrom || host.createdAt,
+            dateTo: args.dateTo || new Date(),
+            timeUnit: args.timeUnit,
+            nodes: resultsToAmountNode(expenseAmountOverTime),
+          };
+
+          const expensesAmountSum = await models.Transaction.sum('amount', { where });
+          const dailyAverageAmount = expensesCount ? Math.abs(expensesAmountSum) / numberOfDays : 0;
+          return {
+            expensesCount,
+            expenseAmountOverTime,
+            invoicesCount: models.Transaction.count({
```

Comment:
I just discovered today that you can instead return functions that will only run if the field is fetched in the query. Let's do that on all these expensive fields :rocket: :rocket: :rocket: 

```suggestion
            invoicesCount: () => models.Transaction.count({
```

Reply:
I've made sure that every field of the `contributionStats` and `expenseStats` are functions. So this should be good now I think. 😉